src/main.rs

```diff
@@ -45,40 +45,208 @@ struct File {
 
-    let file_path = select_stmt.query_row([&uuid], |row| Ok(row.get(0)))?;
-    Ok(file_path?)
+    let mut stmt: rusqlite::Statement<'_> =
+        database.prepare(format!("DELETE FROM {} WHERE uuid = (?1)", TABLE_NAME).as_str())?;
+    stmt.execute([uuid])?;
+
+    Ok(())
 }
 
-async fn return_config_file_get(
-    Path(path): Path<String>,
+fn migrate(redirect_path: String, files: Vec<File>, database: &Connection) -> Response {
+    let migration_target_path = "/tmp/".to_owned() + &uuid::Uuid::new_v4().to_string();
+    match fs::DirBuilder::new().create(&migration_target_path) {
+        Ok(()) => (),
+        Err(e) => {
+            eprint!("Could not create directory: {}", e);
+            return StatusCode::INTERNAL_SERVER_ERROR.into_response();
+        }
+    };
+
+    let log = match migrate_files(&files, migration_target_path.clone()) {
+        Ok(output) => {
+            let log = String::from_utf8_lossy(&output.stderr).to_string();
+
+            if !output.status.success() {
+                return Response::builder()
+                    .status(500)
+                    .header("Content-Type", "text/plain")
+                    .body(format!("The files couldn't be migrated:\n{}", log).into())
+                    .unwrap();
+            }
+            log
+        }
+        Err(e) => {
+            eprint!("Error when migrating files: {}", e);
+            return StatusCode::INTERNAL_SERVER_ERROR.into_response();
+        }
+    };
+
+    let uuid = match add_migration_result_to_db(migration_target_path, log, database) {
+        Ok(uuid) => uuid,
+        Err(_e) => return StatusCode::INTERNAL_SERVER_ERROR.into_response(),
+    };
+    axum::response::Redirect::to(format!("{}/{}", redirect_path, uuid).as_str()).into_response()
+}
+
+fn read_from_db(uuid: String, database: &Connection) -> anyhow::Result<(String, String)> {
+    let mut select_stmt = database.prepare(
+        format!(
+            "SELECT file_path, log from {} WHERE uuid = (?1)",
+            TABLE_NAME
+        )
+        .as_str(),
+    )?;
+
+    let path_log = select_stmt.query_row([&uuid], |row| Ok((row.get(0)?, row.get(1)?)))?;
+    Ok(path_log)
+}
+
+fn generate_json(log: &str, files: Vec<File>) -> String {
+    let mut data = json::JsonValue::new_object();
+    data["log"] = log.into();
+    data["files"] = json::JsonValue::new_array();
+    for file in files {
+        let mut file_data = json::JsonValue::new_object();
+        file_data["fileName"] = file.file_name.into();
+        file_data["fileContent"] = file.file_content.into();
+        data["files"].push(file_data).unwrap();
+    }
+    data.dump()
+}
+
+fn file_arr_from_path(dir_path: String) -> Result<Vec<File>, anyhow::Error> {
+    let mut file_arr: Vec<File> = vec![];
+
+    let dir = match fs::read_dir(dir_path.clone() + "/NM-migrated/system-connections") {
+        Ok(dir) => dir,
+        Err(_e) => fs::read_dir(dir_path + "/NM-migrated")?,
```

Comment:
I don't really get this. The files should definitely always be in the same directory, if that is not the case there is a bug somewhere else and it shouldn't be worked around like this.